lib/ramen/test.js

```diff
@@ -0,0 +1,152 @@
+        analytics.assert(!window.ramenSettings);
+        analytics.assert(!window.Ramen);
+        analytics.initialize();
+        analytics.page();
+        analytics.assert(!window.ramenSettings);
+        analytics.assert(!window.Ramen);
+      });
+
+      it('should call #load', function(){
+        analytics.initialize();
+        analytics.page();
+        analytics.called(ramen.load);
+      });
+    });
+  });
+
+  describe('loading', function(){
+    it('should load', function(done){
+      analytics.load(ramen, done);
+    });
+  });
+
+  describe('after loading', function(){
+    beforeEach(function(done){
+      analytics.once('ready', done);
+      analytics.initialize();
+    });
+
+    describe('#identify', function(){
+      beforeEach(function(){
+        analytics.stub(window.Ramen, 'go');
+      });
+
+      it('should not call Ramen.go if only id is passed', function(){
+        analytics.identify('id');
+        analytics.assert(!window.ramenSettings);
+        analytics.didNotCall(window.Ramen.go);
+
+        analytics.identify('id');
+        analytics.assert(!window.ramenSettings);
+        analytics.didNotCall(window.Ramen.go);
+      });
+
+      it('should call Ramen.go and set correct attributes if just email passed', function(){
+        var email = 'email@example.com';
+        analytics.identify('id', { email: email });
+        analytics.assert(window.ramenSettings.organization_id === '6389149');
+        analytics.assert(window.ramenSettings.user.id === 'id');
+        analytics.assert(window.ramenSettings.user.name === email);
+        analytics.assert(window.ramenSettings.user.email === email);
+        analytics.called(window.Ramen.go);
+      });
+
+      it('should call Ramen.go and set correct attributes if email & name passed', function(){
+        var email = 'email@example.com';
+        var name = 'ryan+segment@ramen.is';
+        analytics.identify('id', { email: email, name: name });
+        analytics.assert(window.ramenSettings.organization_id === '6389149');
+        analytics.assert(window.ramenSettings.user.id === 'id');
+        analytics.assert(window.ramenSettings.user.name === name);
+        analytics.assert(window.ramenSettings.user.email === email);
+        analytics.called(window.Ramen.go);
+      });
+
+      it('should pass along integration options', function(){
+        var email = 'email@example.com';
+        var name = 'ryan+segment@ramen.is';
+        var auth_hash = 'authy_hasy';
+        var timestamp = new Date() / 1000;
+        var custom_links = [{href: 'https://ramen.is/support', title: 'Hello'}];
+        var labels = ['use', 'ramen!'];
+        var environment = 'staging';
+        var logged_in_url = 'https://align.ramen.is/manage';
+        var unknown_future_opt = '11';
+        var unknown_future_user_opt = 'user 11';
+
+        analytics.identify('id', {email: email, name: name},
+          {
+            integrations: {
+              Ramen: {
```

Comment:
Thanks @amillet89. Some comments below.

`environment`: I'm not clear on what you mean when you say "integration-level" and not "call specific." If Segment.com users do not specify the environment here, where else can they specify it? I don't see any mention of environment in the Facade library.

`auth_hash`: Yes. We use `name` and `email` in other parts of our application. `email` is required for our application to work correctly. If `email` is not passed, then the Ramen integration will abort loading. If `email` is passed, but `name` is not, we will set `name` == `email`. Here is a link to our [RamenJS docs](https://ramen.is/pages/ramenjs/js). It is similar to Intercom.

`custom_links`: While these links will tend to be static, there is the possibility that a customer might want to have them be dynamic based on who is logged in, so, unless I'm misunderstanding something, I think it should stay here.

`logged_in_url`: This can be different based on user if, for example, the application uses subdomains, and so I think it is call-specific. However, as I said above I'm unclear on the difference between "integration-level" and "call specific"